internal/services/user_management/user_management.go

```diff
@@ -0,0 +1,108 @@
+package user_management
+
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"Tradeasy/internal/provider/redis"
+	"errors"
+	"strings"
+	"time"
+)
+
+func SignUp(SUpReq SignUpRequest) (SUpRes SignUpResponse, err error) {
+	email := SUpReq.EmailId
+	SUpRes.UserId = strings.Split(email, "@")[0]
+	SUpRes.Password = SUpReq.Password
+	SUpRes.Message = "User registered"
+
+	var user model.Users
+	err = config.DB.Table("users").Where("email_id = ?", SUpReq.EmailId).First(&user).Error
+	if err == nil {
+		return SUpRes, errors.New("email id already registered")
+	}
+	err = config.DB.Table("users").Where("password = ?", SUpRes.Password).First(&user).Error
+	if err == nil {
+		return SUpRes, errors.New("password already taken")
+	}
+	user.UserId = SUpRes.UserId
+	user.EmailId = SUpReq.EmailId
+	user.Password = SUpReq.Password
+	err = config.DB.Table("users").Create(&user).Error
+	if err != nil {
+		return SUpRes, errors.New("signUp failed")
+	}
+	return SUpRes, nil
+}
+
+func UserDetails(detReq UserDetailsRequest, userid string) (detRes UserDetailsResponse, err error) {
+	detRes.TradingAccId = "TRA" + userid
+	detRes.Balance = 0
+	detRes.Message = "User Details registered"
```

Comment:
Do not add message before request gets successfully processed.

Reply:
ok, done